plugin/trino-blackhole/src/main/java/io/trino/plugin/blackhole/BlackHoleSplitManager.java

```diff
@@ -36,6 +37,6 @@ public ConnectorSplitSource getSplits(
             Constraint constraint)
     {
         int splitCount = ((BlackHoleTableHandle) table).splitCount();
-        return new FixedSplitSource(nCopies(splitCount, BlackHoleSplit.INSTANCE));
```

Comment:
what was wrong with nCopies?

Reply:
BTW see https://github.com/trinodb/trino/pull/21633 where i am adding a test (indirectly) exercising the case where connector returns same split instance multiple times. as long as SPI allows that, we need some tests exercising that